### ISSUE TITLE
terminate called after throwing an instance of 'Xbyak::Error' what():  can't protect

### ISSUE THREAD

**Issue by Gvaihir** [mpicbg-csbd/stardist#8]
Hi there! I'm reproducing the training procedure according to your Jupyter with the following configuration

`2019-02-03 20:22:54.463532: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
2019-02-03 20:22:54.478737: I tensorflow/core/common_runtime/process_util.cc:69] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.
number of images: 447
- training:       380
- validation:      67
Configuration for a :class:`StarDist` model.

    Parameters
    ----------
    n_rays : int
        Number of radial directions for the star-convex polygon.
        Recommended to use a power of 2 (default: 32).
    n_channel_in : int
        Number of channels of given input image (default: 1).
    kwargs : dict
        Overwrite (or add) configuration attributes (see below).

    Attributes
    ----------
    unet_n_depth : int
        Number of U-Net resolution levels (down/up-sampling layers).
    unet_kernel_size : (int,int)
        Convolution kernel size for all (U-Net) convolution layers.
    unet_n_filter_base : int
        Number of convolution kernels (feature channels) for first U-Net layer.
        Doubled after each down-sampling layer.
    net_conv_after_unet : int
        Number of extra convolution layers after U-Net (0 to disable).
    train_shape_completion : bool
        Train model to predict complete shapes for partially visible objects at image boundary.
    train_completion_crop : int
        If 'train_shape_completion' is set to True, specify number of pixels to crop at boundary of training patches.
        Should be chosen based on (largest) object sizes.
    train_patch_size : (int,int)
        Size of patches to be cropped from provided training images.
    train_dist_loss : str
        Training loss for star-convex polygon distances ('mse' or 'mae').
    train_epochs : int
        Number of training epochs.
    train_steps_per_epoch : int
        Number of parameter update steps per epoch.
    train_learning_rate : float
        Learning rate for training.
    train_batch_size : int
        Batch size for training.
    train_tensorboard : bool
        Enable TensorBoard for monitoring training progress.
    train_checkpoint : str
        Name of checkpoint file for model weights (only best are saved); set to ``None`` to disable.
    train_reduce_lr : dict
        Parameter :class:`dict` of ReduceLROnPlateau_ callback; set to ``None`` to disable.

        .. _ReduceLROnPlateau: https://keras.io/callbacks/#reducelronplateau
    
Config(n_channel_in=1, n_rays=32, net_conv_after_unet=128, net_input_shape=(None, None, 1), train_batch_size=4, train_checkpoint='weights_best.h5', train_completion_crop=32, train_dist_loss='mae', train_epochs=100, train_learning_rate=0.0003, train_patch_size=(256, 256), train_reduce_lr={'factor': 0.5, 'patience': 10}, train_shape_completion=False, train_steps_per_epoch=400, train_tensorboard=True, unet_kernel_size=(3, 3), unet_n_depth=3, unet_n_filter_base=32)`

And on the epoch 47/100 the process crashes with an error:

`terminate called after throwing an instance of 'Xbyak::Error'
  what():  can't protect
/var/spool/sge/qb3cell/idgpu1/job_scripts/8502: line 34: 55937 Aborted                 (core dumped) python `

or alternatively

`terminate called after throwing an instance of 'Xbyak::Error'
  what():  err munmap
/var/spool/sge/qb3cell/idgpu1/job_scripts/7939: line 34: 21694 Aborted                 (core dumped) python `
Could you suggest any solutions?

Thanks ahead!

**Comment by uschmidt83** [comment]
Hi @Gvaihir, thanks for trying out our code!
Unfortunately, I'm afraid I cannot help you with this problems. It seems unrelated to `stardist`...

Best,
Uwe

**Comment by Gvaihir** [comment]
Thanks! Could you post if stardist depends on any specific tensor flow version?

**Comment by uschmidt83** [comment]
No, StarDist doesn't require a specific version of TensorFlow.

We implicitly (through [csbdeep](https://pypi.org/project/csbdeep/)>=0.3.0 dependency) require Keras version 2.1.2 or later. So any version of TensorFlow that is compatible with your installed version of Keras should be fine.

**Comment by Gvaihir** [comment]
Thanks, Uwe! 
Closing the Issue!
Cheers,
G

**Comment by SaiSatwik** [comment]
@Gvaihir May I know, how did you resolve this issue? Because I am getting the same error.

**Comment by Gvaihir** [comment]
@SaiSatwik I think I just used latest CUDA driver and updated TF. Sorry, was very long time ago.
